uber/site_sections/registration.py

```diff
@@ -416,11 +416,12 @@ def register(self, session, message='', **params):
                     message = 'Please enter a real emergency contact number'
                 elif attendee.age_group == AGE_UNKNOWN:
                     message = 'Please select an age category'
-                elif attendee.payment_method == MANUAL and not attendee.email:
+                elif attendee.payment_method == MANUAL and not re.match(EMAIL_RE, attendee.email):
                     message = 'Email address is required to pay with a credit card at our registration desk'
                 elif attendee.badge_type not in [ATTENDEE_BADGE, ONE_DAY_BADGE]:
                     message = 'No hacking allowed!'
                 else:
```

Comment:
@kitsuta moved this in an earlier commit:
https://github.com/magfest/ubersystem/pull/980/files#diff-1d4b74d1828afc36409f8101ea8a5f15L410
and I didn't catch the error when reviewing the pull request.

The problem is that there's an ``HTTPRedirect`` if the payement method is Stripe, but that means that we never added the attendee to the session before redirecting, which is why they weren't being saved.

To clarify the semantics, the two following are identical:

```python
foo.bar = 'baz'
session.add(foo)
session.commit()  # works as you'd expect

session.add(foo)
foo.bar = 'baz'
session.commit()  # this does the same thing as the above block
```

So it doesn't actually matter whether you modify something before or after it's added to the session, saying ``session.add(whatever)`` just means that ``whatever`` will be persisted to the database whenever the session is committed.  So it's okay to add it at the top of this if-block even though we're mutating the ``attendee`` after adding it.